### PR TITLE
add possibility to find overlapping matches

### PR DESCRIPTION
**Pull request by adamiwaniuk**
Feature from issue: https://github.com/codilime/veles/issues/240

### CODE REVIEW

File: src/ui/searchdialog.cc [hunk 1]
```diff
@@ -118,11 +118,16 @@ qint64 SearchDialog::findNext() {
   }
 
   bool backwards = ui->cbBackwards->isChecked();
+  bool overlapping = ui->cbOveralpping->isChecked();
```
Comment (mkow) [inline src/ui/searchdialog.cc]: fix typo in `cbOveralpping`

File: src/ui/searchdialog.ui [hunk 2]
```diff
@@ -98,21 +98,36 @@
...
+              </property>
+              <property name="text">
+               <string>&amp;Prompt on replace</string>
+              </property>
+             </widget>
+            </item>
+           </layout>
+          </item>
+          <item>
+           <widget class="QCheckBox" name="cbOveralpping">
```
Comment (mkow) [inline src/ui/searchdialog.ui]: typo in cbOveralpping

File: src/ui/searchdialog.ui [hunk 3]
```diff
@@ -185,6 +200,7 @@
   <tabstop>cbReplace</tabstop>
   <tabstop>cbBackwards</tabstop>
   <tabstop>cbPrompt</tabstop>
+  <tabstop>cbOveralpping</tabstop>
```
Comment (mkow) [inline src/ui/searchdialog.ui]: typo in `cbOveralpping`

**Comment by adamiwaniuk** [comment]
In my opinion saving option like that in settings is not consistent with options like that in most of applications

**Comment by mkow** [comment]
Another bug: Searching backwards currently doesn't care about the checkbox.

File: src/ui/searchdialog.cc [hunk 4]
```diff
@@ -120,16 +120,21 @@ qint64 SearchDialog::findNext() {
   bool backwards = ui->cbBackwards->isChecked();
   bool overlapping = ui->cbOverlapping->isChecked();
 
-  qint64 startSearchPos = _lastFoundPos;
-
-  if (!backwards) {
-    if (!overlapping) {
-      startSearchPos += _lastFoundSize;
+  qint64 start_search_pos_modifer = 0;
```
Comment (mkow) [inline src/ui/searchdialog.cc]: `modifer` -> `modifier`

File: src/ui/searchdialog.cc [hunk 5]
```diff
@@ -120,16 +120,21 @@ qint64 SearchDialog::findNext() {
...
-      startSearchPos += 1;
+      start_search_pos_modifer = _lastFoundSize;
     }
   }
 
+  if (backwards) {
+    start_search_pos_modifer = -(start_search_pos_modifer - 1);
+  }
+
+  qint64 startSearchPos = _lastFoundPos + start_search_pos_modifer;
```
Comment (mkow) [inline src/ui/searchdialog.cc]: `startSearchPos` -> `start_search_pos`

**Pull request merged**